src/components/MenuOverlay/useMenuNavigation.ts

```diff
@@ -34,6 +35,7 @@ export const userNavigations: Array<NavigationObject> = [
 ];
 
 export const useMenuNavigation = (role: Role | undefined) => {
+  const superAdminHomePageContext = useSuperAdminHomePageFunction();
   const adminGalleryContext = useAdminGalleryFunction();
```

Comment:
Please destructure where possible, it's cleaner code in my opinion.

Reply:
I destruct  useSuperAdminHomePageFunction, for that I had to make some changes in  superAdminHomePageContext. 

useAdminGalleryFunction() I don't want to touch in this branch. it has nothing to do with the meanig of this brach.